### PR TITLE
alphabetize the keys in dependencies and devDependencies. (for addon creation)

### PR DESCRIPTION
**Pull request by kellyselden**
JSON.stringify in this case is resulting in out of order dependencies.
The next `npm install --save-dev` for a user will screw up their diffs:

capture

This will keep the files ordered.

Also puts back the newline that gets removed in the stringify.

### CODE REVIEW

**Comment by kellyselden** [comment]
This could probably use a test...

**Comment by kellyselden** [comment]
I've added tests, and this is ready for review. I've tested locally on Windows, so any CI issues I "think" are pre-existing?

File: blueprints/addon/index.js [hunk 1]
```diff
@@ -52,7 +55,16 @@ module.exports = {
...
 
-    fs.writeFileSync(path.join(this.path, 'files', 'bower.json'), JSON.stringify(contents, null, 2));
+    this._writeContentsToFile(contents, 'bower.json');
+  },
+
+  _writeContentsToFile: function(contents, fileName) {
+    fs.writeFileSync(path.join(this.path, 'files', fileName), JSON.stringify(contents, null, 2) + '\n');
+  },
+
+  _alphabetizeDependencies: function(contents) {
```
Comment (stefanpenner) [inline blueprints/addon/index.js]: this method doesnt use the classes internal state, it should not be on the prototype.

File: blueprints/addon/index.js [hunk 2]
```diff
@@ -52,7 +55,16 @@ module.exports = {
 
     contents.name = this.project.name();
 
-    fs.writeFileSync(path.join(this.path, 'files', 'bower.json'), JSON.stringify(contents, null, 2));
+    this._writeContentsToFile(contents, 'bower.json');
+  },
+
+  _writeContentsToFile: function(contents, fileName) {
+    fs.writeFileSync(path.join(this.path, 'files', fileName), JSON.stringify(contents, null, 2) + '\n');
```
Comment (stefanpenner) [inline blueprints/addon/index.js]: i believe instead https://github.com/substack/json-stable-stringify could be used.

File: blueprints/addon/index.js (hunk 1 above)
Comment (kellyselden) [inline blueprints/addon/index.js]: agreed

File: blueprints/addon/index.js (hunk 2 above)
Comment (kellyselden) [inline blueprints/addon/index.js]: I thought there was a reason I didn't/couldn't use that, but it's escaping me. I will look into it again.

Comment (stefanpenner) [inline blueprints/addon/index.js]: Ya its possible I missed something.

**Comment by kellyselden** [comment]
comments addressed. `json-stable-stringify` worked like a charm!

File: blueprints/addon/index.js [hunk 3]
```diff
@@ -149,3 +154,11 @@ module.exports = {
     return entityName;
   }
 };
+
+function alphabetizeObjectKeys(unordered) {
```
Comment (stefanpenner) [inline blueprints/addon/index.js]: This I think is no longer needed

**Comment by kellyselden** [comment]
I remember why I didn't use `json-stable-stringify` now. It alphabetizes the whole file, putting `dependencies` up top above `scripts`. I think I have to revert.

**Comment by stefanpenner** [comment]
> I think I have to revert.

ok

**Comment by kellyselden** [comment]
This should be good now. I had to add lots of unit tests because there were none previously. The failures all have to do with EADDRINUSE, which I don't believe I cause. Tests work locally.

**Comment by rwjblue** [comment]
Looks like some JSHint failures are all that is left.

**Comment by kellyselden** [comment]
ahh forgot the `/*jshint multistr: true */`

**Comment by kellyselden** [comment]
looking green now.

**Comment by kellyselden** [comment]
@homu r+
